src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
+{
+    public class LinuxContainerInitializationHostService : IHostedService
+    {
+        private readonly ScriptSettingsManager _settingsManager;
+        private readonly IInstanceManager _instanceManager;
+        private readonly ILogger _logger;
+        private CancellationToken _cancellationToken;
+
+        public LinuxContainerInitializationHostService(ScriptSettingsManager settingsManager, IInstanceManager instanceManager, ILoggerFactory loggerFactory)
+        {
+            _settingsManager = settingsManager;
+            _instanceManager = instanceManager;
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Initializing LinuxContainerInitializationService.");
+            _cancellationToken = cancellationToken;
+            await Run();
+        }
+
+        private async Task Run()
+        {
+            if (_settingsManager.IsLinuxContainerEnvironment)
+            {
+                await InitializeAssignmentContext();
+            }
+        }
+
+        private async Task InitializeAssignmentContext()
+        {
+            var startContext = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerStartContext);
+
+            // Container start context is not available directly
+            if (string.IsNullOrEmpty(startContext))
+            {
+                _logger.LogInformation("AssignmentContext not available in ContainerStartContext");
+                // Check if the context is available in blob
+                var sasUri = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerStartContextSasUri);
+
+                if (!string.IsNullOrEmpty(sasUri))
+                {
+                    _logger.LogInformation("AssignmentContext ContainerStartContextSasUri available");
+                    startContext = await GetAssignmentContextFromSasUri(sasUri);
+                }
+            }
+
+            if (!string.IsNullOrEmpty(startContext))
+            {
+                _logger.LogInformation("Assigning HostAssignmentContext.");
+
+                var encryptedAssignmentContext = JsonConvert.DeserializeObject<EncryptedHostAssignmentContext>(startContext);
+                var containerKey = _settingsManager.GetSetting(EnvironmentSettingNames.ContainerEncryptionKey);
+                var assignmentContext = encryptedAssignmentContext.Decrypt(containerKey);
+                if (_instanceManager.StartAssignment(assignmentContext))
+                {
+                    _logger.LogInformation("Assign HostAssignmentContext success");
```

Comment:
These log statements are a little misleading since the actual assignment is fire and forget and runs in the background. I'd just remove these additional log statements since StartAssignment already logs.

Reply:
Yes. Most of the work is in the background thread. but StartAssignment also validates the assignmentcontext hasn't changed for the container from before.
Will change the log to "Start assign HostAssignmentContext success"